common/persistence/persistenceTestBase.go

```diff
@@ -111,83 +114,110 @@ func (g *testTransferTaskIDGenerator) GetNextTransferTaskID() (int64, error) {
-		log)
-	if err != nil {
-		log.Fatal(err)
-	}
+		s.MetadataManager, err = NewCassandraMetadataPersistence(options.ClusterHost, options.ClusterPort, options.ClusterUser,
+			options.ClusterPassword, options.Datacenter, s.CassandraTestCluster.keyspace, currentClusterName, log)
+		if err != nil {
+			log.Fatal(err)
+		}
 
-	s.HistoryMgr, err = NewCassandraHistoryPersistence(options.ClusterHost, options.ClusterPort, options.ClusterUser,
-		options.ClusterPassword, options.Datacenter, s.CassandraTestCluster.keyspace, 2, log)
-	if err != nil {
-		log.Fatal(err)
-	}
+		s.VisibilityMgr, err = NewCassandraVisibilityPersistence(options.ClusterHost, options.ClusterPort,
+			options.ClusterUser, options.ClusterPassword, options.Datacenter, s.CassandraTestCluster.keyspace, log)
+		if err != nil {
+			log.Fatal(err)
+		}
 
-	s.MetadataManager, err = NewCassandraMetadataPersistence(options.ClusterHost, options.ClusterPort, options.ClusterUser,
-		options.ClusterPassword, options.Datacenter, s.CassandraTestCluster.keyspace, currentClusterName, log)
-	if err != nil {
-		log.Fatal(err)
-	}
+		s.TaskIDGenerator = &testTransferTaskIDGenerator{}
+
+		// Create a shard for test
+		s.readLevel = 0
+		s.replicationReadLevel = 0
+		s.ShardInfo = &ShardInfo{
+			ShardID:                 shardID,
+			RangeID:                 0,
+			TransferAckLevel:        0,
+			ReplicationAckLevel:     0,
+			TimerAckLevel:           time.Time{},
+			ClusterTimerAckLevel:    map[string]time.Time{currentClusterName: time.Time{}},
+			ClusterTransferAckLevel: map[string]int64{currentClusterName: 0},
+		}
 
-	s.VisibilityMgr, err = NewCassandraVisibilityPersistence(options.ClusterHost, options.ClusterPort,
-		options.ClusterUser, options.ClusterPassword, options.Datacenter, s.CassandraTestCluster.keyspace, log)
-	if err != nil {
-		log.Fatal(err)
-	}
+		err1 := s.ShardMgr.CreateShard(&CreateShardRequest{
+			ShardInfo: s.ShardInfo,
+		})
+		if err1 != nil {
+			log.Fatal(err1)
+		}
+	} else {
+		var err error
+		s.MetadataManager, err = NewMysqlMetadataPersistence("uber",
+			"uber",
+			"localhost",
+			"3306",
+			"catalyst_test")
 
-	s.TaskIDGenerator = &testTransferTaskIDGenerator{}
-
-	// Create a shard for test
-	s.readLevel = 0
-	s.replicationReadLevel = 0
-	s.ShardInfo = &ShardInfo{
-		ShardID:                 shardID,
-		RangeID:                 0,
-		TransferAckLevel:        0,
-		ReplicationAckLevel:     0,
-		TimerAckLevel:           time.Time{},
-		ClusterTimerAckLevel:    map[string]time.Time{currentClusterName: time.Time{}},
-		ClusterTransferAckLevel: map[string]int64{currentClusterName: 0},
-	}
+		if err != nil {
+			log.Fatal(err)
+		}
 
-	err1 := s.ShardMgr.CreateShard(&CreateShardRequest{
-		ShardInfo: s.ShardInfo,
-	})
-	if err1 != nil {
-		log.Fatal(err1)
+		db, err := sqlx.Connect("mysql",
+			"uber:uber@tcp(localhost:3306)/catalyst_test")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		file, err := ioutil.ReadFile("./sql/domains.sql")
```

Comment:
This should be handled by an upgrade lib.

Reply:
Deferred to next PR